storage/client/queries/queries.go

```diff
@@ -147,46 +147,62 @@ const (
+			ORDER BY address -- TODO: maybe order by account balance?
+			LIMIT $2::bigint
+			OFFSET $3::bigint
+		),
+		-- Filtered accounts delegations delegatee's account infos.
+		active_delegations AS (
+			SELECT
+				d.delegator,
+				d.shares,
+				a.escrow_balance_active,
+				a.escrow_total_shares_active
+			FROM chain.delegations d
+			JOIN chain.accounts a ON d.delegatee = da.address
+			WHERE
+				d.delegator IN (SELECT address FROM filtered_accounts)
+		),
+		-- Filtered accounts debonding delegations delegatee's account infos.
+		debonding_delegations AS (
+			SELECT
+				dd.delegator,
+				dd.shares,
+				a.escrow_balance_debonding,
+				a.escrow_total_shares_debonding
+			FROM chain.debonding_delegations dd
+			JOIN chain.accounts a ON dd.delegatee = dda.address
+			WHERE
+				dd.delegator IN (SELECT address FROM filtered_accounts)
+		)
 		SELECT
-			address,
-			COALESCE(nonce, 0),
-			COALESCE(general_balance, 0),
-			COALESCE(escrow_balance_active, 0),
-			COALESCE(escrow_balance_debonding, 0)
-		FROM chain.accounts
-		WHERE ($1::numeric IS NULL OR general_balance >= $1::numeric) AND
-					($2::numeric IS NULL OR general_balance <= $2::numeric) AND
-					($3::numeric IS NULL OR escrow_balance_active >= $3::numeric) AND
-					($4::numeric IS NULL OR escrow_balance_active <= $4::numeric) AND
-					($5::numeric IS NULL OR escrow_balance_debonding >= $5::numeric) AND
-					($6::numeric IS NULL OR escrow_balance_debonding <= $6::numeric) AND
-					($7::numeric IS NULL OR general_balance + escrow_balance_active + escrow_balance_debonding >= $7::numeric) AND
-					($8::numeric IS NULL OR general_balance + escrow_balance_active + escrow_balance_debonding <= $8::numeric)
-		ORDER BY address
-		LIMIT $9::bigint
-		OFFSET $10::bigint`
-
-	Account = `
-		SELECT
-			address,
-			COALESCE(nonce, 0),
-			COALESCE(general_balance, 0),
-			COALESCE(escrow_balance_active, 0),
-			COALESCE(escrow_balance_debonding, 0),
-			COALESCE (
-				(SELECT COALESCE(ROUND(SUM(shares * escrow_balance_active / escrow_total_shares_active)), 0) AS delegations_balance
-				FROM chain.delegations
-				JOIN chain.accounts ON chain.accounts.address = chain.delegations.delegatee
-				WHERE delegator = $1::text AND escrow_total_shares_active != 0)
-			, 0) AS delegations_balance,
-			COALESCE (
-				(SELECT COALESCE(ROUND(SUM(shares * escrow_balance_debonding / escrow_total_shares_debonding)), 0) AS debonding_delegations_balance
-				FROM chain.debonding_delegations
-				JOIN chain.accounts ON chain.accounts.address = chain.debonding_delegations.delegatee
-				WHERE delegator = $1::text AND escrow_total_shares_debonding != 0)
-			, 0) AS debonding_delegations_balance
-		FROM chain.accounts
-		WHERE address = $1::text`
+			fa.address,
+			fa.nonce,
+			fa.general_balance,
+			fa.escrow_balance_active,
+			fa.escrow_balance_debonding,
+			COALESCE(ROUND(SUM(ad.shares * ad.escrow_balance_active / NULLIF(ad.escrow_total_shares_active, 0))), 0) AS delegations_balance,
```

Comment:
interesting, I also wonder about pushing this aggregation into the delegations subquery

Reply:
Hm yeah i think doing that makes sense

Reply:
Done. This query was now moved into the meterialized view.